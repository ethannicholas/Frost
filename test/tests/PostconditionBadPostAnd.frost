@extendable
class PostconditionBadPost1 {
    var x := 0

    @extendable
    @postAnd(x > 0)
    method test(x:Int) {
        self.x := x
    }
}

class PostconditionBadPost2 : PostconditionBadPost1 {
    @override
    @postAnd(x > 0)
    method test(x:Int) {
        self.x := x
    }
}