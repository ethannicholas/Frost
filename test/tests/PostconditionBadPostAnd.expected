PostconditionBadPostAnd.frost:4:16: error: cannot use '@postAnd' in a method which does not override another method; did you mean '@post'?
1 error
